ReadMe
***************
NitrOS-9 ReadMe
***************

Welcome to NitrOS-9!

DISTRIBUTION
============
There are two double-sided 40 track DSK images ending in '_ds40_1.dsk',
and '_ds40_2.dsk' for disks 1 and 2, respectively.  For those with
3.5" 720K drives, an 80 track DSK image ending in '_ds80.dsk' is also
included in the distribution.

USING THE DISK IMAGES
=====================
If you wish to transfer the disk images to a floppy disk for use with a
real CoCo, follow the steps below:

WINDOWS/DOS:

1. Download and install the latest version of DSKINI/RETRIEVE at
   http://www.nitros9.org/

2. Insert a blank double-sided floppy in to your drive (we assume B:)

3. Type the following command, replacing the name of the image:

             DSKINI /T40 /D B: <diskimagefile>     (360K 5.25" floppy disk)
             DSKINI /T80 /D B: <diskimagefile>     (720K 3.5"  floppy disk)

   Where <diskimagefile> is the image file you want to transfer to the
   floppy disk.  The image file transfer should start.  When it is done,
   take out the disk and try booting your CoCo with it.

Repeat steps 2-3 for each additional disk.


LINUX:

1. Download and install the latest version of fdutils from
   http://fdutils.linux.lu/

2. Once installed, edit the /usr/local/etc/mediaprm file and add the
   following lines at the end of the file:

     "COCO360":
      DS DD sect=18 cyl=40 ssize=256 tpi=48

     "COCO720":
      DS DD sect=18 cyl=80 ssize=256 tpi=96

3. Insert your blank floppy into the drive on your Linux box.

4. Type the following command:

             setfdprm /dev/fd1 coco360    (360K 5.25" floppy disk)
             setfdprm /dev/fd1 coco720    (720K 3.5"  floppy disk)

   This assumes your floppy drive is /dev/fd1.  You may need to make
   adjustments to the above commands for your environment.

5. After typing the setfdprm command, the floppy drive unit should spin
   for a second then stop. Type this command:

             dd if=<diskimagefile> of=/dev/fd1

   Where <diskimagefile> is the image file you want to transfer to the
   floppy disk.  The image file transfer should start.  When it is done,
   take out the disk and try booting your CoCo with it.

Repeat steps 3-5 for each additional disk.

Addendum by Gene Heskett

Unfortunately, many of the fdc chipsets shipping on motherboards for
the last 5 years or so, do NOT understand, and will probably crash your
pc when a 256 byte/sector disk read or write is attempted.  This rather
high priced Asus M2n-SLI Deluxe board is one such board, so using floppies
for the 'sneakernet' to get this stuff onto my coco3 cannot be done.
email to gheskett AT wdtv DOT com.

1. This repo is newer than the sourceforge repo, and a lot more of it
Just Works(TM).  If you find something that doesn't, use the above
email address.

2. This repo's work will be transfered back to sourceforge when I can
figure out how to do a commit in mercurial.  Tutorials on this are welcome.
There are, since I did the hg clone about the 27th of October 2012, 
probably at least 50 files needing committed, changed in getting this
back to buildable and bootable.

3. Since we now have drivewire, the appropriate for your machine .dsk
files can be downloaded and mounted directly as a fully read/write
virtual disk.  Drivewire is the cats meow, many thanks to Aaron Wolfe.
Make yourself a cable and join in on the dance floor, its amazing.

4. I am still using mamou as the assembler, no attempt has been made
to convert it to using William Astles lwtools just yet as I wanted
this to be as close to 100% usable as possible before attempting the
conversion so that only the conversion generated errors remain.  There
will be some teething problems while that is going on in a separate
tree that won't be visible until it Just Works(TM).

5. One thing that was recently done by Boisy was to convert the
boot_scsi.asm file from, because of code size constraints, the use of 
the former drive addressing scheme of a single marching set bit across 
the scsi bus drive address field, into a base zero decimal scheme, so the
drive address contained in the 4th byte from the end can now be looked
at as a decimal value where $00 is now a legal address for drive 0 on
the bus.  A MAJOR change, and one in the right direction IMO, that was
not even mentioned in the comments of the boot_scsi.asm header till I
discovered it yesterday while reading the code trying to figure out why
with that byte set to $01 like it had been for years, the boot was 
failing because it was trying to find OS9Boot on my drive 1, s1, and 
fixed the makefiles accordingly.

6. I am not trying to keep this a secret but it probably will not get
any ink on the coco list until such time as I have committed to source-
forge and that repo is again useable, so bear with me.  I have cable
dsl so my 'intertube' at 3 megabits isn't that fast.  A lot of traffic 
will probably result in my isp blocking the port address I do use to
get past the port 80 block just like they do without admitting to it 
because if they did admit it, the FCC would probably discuss their 
common carrier status with the ISP's legal dept.  As that status has
huge tax and licensing advantages to the ISP, they will not give that
status up without a prolonged court fight.  I have even considered
putting this page on port 6809, but when and if that happens, you'll
see it in my email sig.

Thats all for now folks, keep the old girl running till the rapture
or whatever.

Cheers, Gene


ChangeLog
Please consult the NitrOS-9 source repository for a complete
list of changes: https://sourceforge.net/p/nitros9/code/

In a local repository checkout, the full log can be viewed
with the following command: hg log -v

----------------------------------------------------------

NitrOS-9 upcoming release:

- Level 1 6309 support (Robert Gault, Tormod Volden)
- Native CoCoSDC boot module (Darren Atkinson)
- sdir and sdrive tools for CoCoSDC (Tim Lindner)
- rbsuper descriptor fix (Christopher Hawks)
- cobbler: fragmented OS9Boot support (Robert Gault)
- deblocking IDE driver (David Ladd)
- Optional timeout for Becker port (David Ladd)
- brainfuck interpreter (Boisy Pitre)
- grfdrv line drawing improvements (Robert Gault)
- sierra text rectangle fix (Guillaume Major)
- Build with lwtools 4.12 (William Astle)
- Various build/makefile cleanups (Gene Heskett, Tormod Volden)

----------------------------------------------------------

NitrOS-9 V3.3.0:

- CoHR: Now supports COVDG control codes (Ken H.)
- New MinTED - Minimalist Text Editor (Luis Antoniosi)
- rb1773: Make 48 TPI floppy disks readonly in 96 TPI drives
  (Robert Gault and Gene Heskett)
- rbdesc/superdesc: Raise default SAS from $8 to $10 (Gene)
- Gene's myram RAM disk driver renamed to MRAM and included
  on L2 disk images
- Dragon: Working disk images again (Tormod Volden)
- Fixed ROM boot images (Tormod)
- Added CoCo SDC driver (Boisy Pitre)
- Added CoCo Deluxe port (Boisy)

3rdparty:
- King's Quest 4: Many updates (Guillaume Major)
- New utils disk with Gene's bootlink and vfy utilities
- Pacos9: Brough back to life (Tormod)

General:
- Major cleanup of Makefiles and build infrastructure (Tormod)
- Everything now builds with lwtools (Tormod)
- Most (all?) things in shape again after lwtools transition
  (Tormod, David Ladd, Bob Devries)

----------------------------------------------------------

NitrOS-9 V3.2.9:

This version was not officially prepared and released. However,
many development builds (from both default and lwtools branch)
have been floating around with this version, so to avoid confusion,
the next release will have the version bumped. 

----------------------------------------------------------
2013/01/29 David Ladd
M ChangeLog
M level1/coco1/modules/makefile
M level2/coco3/modules/makefile
M rules.mak

Updated rules.mak to include values that should have been set
for use in the level1/coco1/modules/makefile &
level2/coco3/modules/makefile.

Removed added values that aren't needed as those are now set
in the rules.mak file.

----------------------------------------------------------
2013/01/28 David Ladd
M ChangeLog
M level1/coco1/modules/makefile
M level1/modules/rb1773desc.asm
M level1/modules/rbdesc.asm
M level1/modules/rfmdesc.asm
M level1/modules/superdesc.asm
M level2/coco3/modules/makefile

Updated level1/coco1/modules/makefile & level2/coco3/modules/makefile
so that correct values would be sent to assembler when
building superdesc.asm for s(x).dd and i(x).dd descripters.


----------------------------------------------------------
2013/01/28 David Ladd
M ChangeLog
M level1/modules/ddiskdesc.asm
M level1/modules/dwdesc.asm
M level1/modules/rb1773desc.asm
M level1/modules/scdwvdesc.asm

Updated level1/modules/ddiskdesc.asm so that if default
values are not given by assembler define command statement
then definitions are set to a default value.

Updated level1/modules/rb1773desc.asm so that if default
values are not given by assembler define command statement
then definitions are set to a default value.

----------------------------------------------------------
2013/01/28 David Ladd
M ChangeLog
M level2/coco3/bootfiles/makefile
M level2/coco3/makefile

Updated level2/coco3/bootfiles/makefile to create bootfiles
for headless becker and headless arduino.

Updated level2/coco3/makefile to create boot disk images
for headless becker and headless arduino.

----------------------------------------------------------
2013/01/28 David Ladd
M ChangeLog
M level2/coco3/bootfiles/makefile

Updated level2/coco3/bootfiles/makefile so that the becker
os9boot file has ddx0.dd rather then ddd0.dd.

----------------------------------------------------------
2013/01/27 David Ladd
M ChangeLog
M level2/coco3/cmds/makefile

Updated level2/coco3/cmds/makefile to detect for H6309
and if set use the sys6309l2.a lib file otherwise
use sys6809l2.a lib file.

----------------------------------------------------------
2013/01/27 David Ladd
M ChangeLog
M level1/coco1/bootfiles/makefile
M level1/coco1/modules/makefile

Updated level1/coco1/modules/makefile so that boot_dw_coco1
and dwio_coco1.sb get created.

Updated level1/coco1/bootfiles/makefile added detection of
PORT status so if coco1 is detected then dwio_coco1.sb and
boot_dw_coco1 are used in place of dwio.sb and boot_dw.

----------------------------------------------------------
2013/01/26 David Ladd
M ChangeLog
M level1/coco1/bootfiles/makefile
M level1/coco1/makefile
M level1/coco1/modules/makefile

Updated level1/coco1/bootfiles/makefile to create becker &
arduino boot and kernel files.

Updated level1/coco1/makefile to create becker & arduino
boot disks for use with drivewire server.

Updated level1/coco1/modules/makefile to create becker &
arduino dwio sb files.

----------------------------------------------------------
2013/01/25 David Ladd
M 3rdparty/packages/sierra/kingsquest2/makefile
M ChangeLog
M level2/coco3/makefile

Updated "3rdparty/packages/sierra/kingsquest2/makefile" to
fix issue with not using the correct dwio.sb and
dwio_becker.sb.

In "level2/coco3/makefile" updated from
"LDSKBECKER      = $(DISTRONAME)_80d_becker.dsk" to 
"LDSKBECKER      = $(DISTRONAME)_becker.dsk"
since the becker boot disks are for use in the drivewire
server and those are not double sided disks.

----------------------------------------------------------
2013/01/23 David Ladd
M 3rdparty/utils/view/makefile
M ChangeLog
M level1/modules/dwread.asm
M level1/modules/dwwrite.asm


defs/rbf.d
               IFNE      RBF.D-1

RBF.D          SET       1

********************************************************************
* rbf.d - Random Block File Manager Definitions
*
* $Id$
*
* RBF stands for 'Random Block Filemanager' and is a package of subroutines
* that define the logical structure of a disk and allows access to the files
* in that structure.
*
* The data structures in this file give RBF its 'personality' and are used
* by RBF itself, as well as applications that will require disk I/O.
*
* Edt/Rev  YYYY/MM/DD  Modified by
* Comment
* ------------------------------------------------------------------
*          1982/07/13  Robert F. Doggett
* PD.Exten added to path descriptor and PE Entries defined.
*
*          1982/07/15  Robert F. Doggett
* V.FileHd inserted in driver static storage.
*
*          1982/09/10  WGP
* Level One/Level Two condionals added.
*
*          1982/09/17  WGP
*
*          1982/09/17  Robert F. Doggett
* PD.SLE renamed to PD.Creat.
*
*          1982/09/17  Robert F. Doggett
* V.DiskID, V.BMapSz, V.MapSct added for smart multi-sector bitmap searching.
*
*          1982/09/20  Robert F. Doggett
* Reserved areas added in static storage.

*          1983/06/07  Robert F. Doggett
* Added InDriver flag in PD.SMF.
*
*          1983/06/13  Robert F. Doggett
* Added PE.Req tmp save for PE.Lock.
*
*          1983/08/08  Robert F. Doggett
* Reserved PD.SToff for Japanese.
*
*          1983/11/19  Robert F. Doggett
* Added V.ResBit in drive tables.
*
*          1983/12/12  Robert F. Doggett
* Added PE.Prior to save process priority.
*
*          1983/12/13  Robert F. Doggett
* Added BufBusy bit in state flag (PD.SMF).
*
*          1984/07/06  Mark G. Hawkins
* Added Bit Definitions for DD.FMT.
*
*          1987/06/23  Kevin K. Darling
* Updated with Dev Desc Info
*
*          2003/06/02  Boisy G. Pitre
* Updated comments for clarity.
* Added TYP.256-TYP.2048 definitions for future expansion.
*
*          2004/04/14  Boisy G. Pitre
* Added new entries to device descriptor as per OS-9 Level One V2.00.00
* addendum.  Pointed out by Rodney Hamilton.
*
*	       2005/05/31  P.Harvey-Smith.
* Added defines for bit 2 of IT.DNS, which the format command uses to 
* determine if a drive should have it's track 0 as single or double density
*
*          2005/11/23  Robert Gault
* Added IT.MPI giving it duplicate location with IT.TFM. This will not be
* transferred to the path descriptor, thus the  equ *  rahter than  rmb 1.
*
*          2005/12/11  Boisy G. Pitre
* Added IT.SOFF1-ITSOFF3 and IT.LLDRV for SuperDrivers.

               NAM       rbf.d
               TTL       Random Block File Manager Definitions

               PAG       
*******************************
* RBF Device Descriptor Offsets
*
* These definitions are for RBF device descriptors.

               ORG       M$DTyp
               RMB       1                   Device type (DT.RBF)
IT.DRV         RMB       1                   Drive number
IT.STP         RMB       1                   Step rate
IT.TYP         RMB       1                   Disk device type information
IT.DNS         RMB       1                   Density capability
IT.CYL         RMB       2                   Number of cylinders
IT.SID         RMB       1                   Number of surfaces
IT.VFY         RMB       1                   Verify disk writes (0 = verify, 1 = don't)
IT.SCT         RMB       2                   Default sectors/track
IT.T0S         RMB       2                   Default sectors/track for track 0 sector 0
IT.ILV         RMB       1                   Sector interleave offset
IT.SAS         RMB       1                   Segment allocation size
* The following fields are from the OS-9 Level One V2.00.00 Addendum
IT.TFM         RMB       1                   DMA Transfer Mode
IT.Exten       RMB       2                   Path Extension (PE) for record locking
IT.SToff       RMB       1                   Sector/Track offsets (for "foreign" disk formats)
* The following fields are not copied to the path descriptor
IT.WPC         RMB       1                   Write precomp cyl/4 (HD)
IT.OFS         RMB       2                   Starting cylinder offset (HD)
IT.RWC         RMB       2                   Reduced write current cylinder (HD)
* These fields have been added because of SuperDriver.  They probably
* can be used in other drivers
               ORG       IT.WPC
IT.SOFF1       RMB       1
IT.SOFF2       RMB       1
IT.SOFF3       RMB       1
IT.LLDRV       RMB       2
IT.MPI         RMB       1


* IT.TYP Definitions
*
TYP.HARD       EQU       %10000000           Hard disk
TYP.FLP        EQU       %00000000           Floppy disk
TYP.NSF        EQU       %01000000           Non-standard format
TYP.SOF        EQU       %00000000           Standard NitrOS-9 format

* IT.TYP bit definitions if bit 7 of IT.TYP is set (hard disk)
TYPH.256       EQU       %00000000           256 byte sector media
TYPH.512       EQU       %00000001           512 byte sector media
TYPH.1024      EQU       %00000010           1024 byte sector media
TYPH.2048      EQU       %00000011           2048 byte sector media
TYPH.SSM       EQU       %00000011           Sector size mask
TYPH.DRSV      EQU       %00001100           Driver-reserved bits
TYPH.DSQ       EQU       %00010000           Drive size query flag

* IT.TYP bit definitions if bit 7 of IT.TYP is clear (floppy disk)
TYP.5          EQU       %00000000           5" media
TYP.3          EQU       %00000001           3.5" media
TYP.SBO        EQU       %00000010           Sector base offset (clear = 0; sect = 1)
TYP.256        EQU       %00000000           256 byte sector media
TYP.512        EQU       %00000100           512 byte sector media
TYP.CCF        EQU       %00100000           CoCo format
TYP.NCCF       EQU       %00000000           Non-CoCo format

* IT.DNS Definitions
*

* If bit 7 of IT.TYP is set, IT.DNS is driver-dependent

* IT.DNS bit definitions if bit 7 of IT.TYP is clear (floppy disk)
DNS.FM         EQU       %00000000           Single-density (FM)
DNS.MFM        EQU       %00000001           Double-density (MFM)
DNS.STD        EQU       %00000000           Single track distance (48/135 tpi)
DNS.DTD        EQU       %00000010           Double track distance (96 tpi)

* Added PHS,2005-05-31, as format seems to use these
DNS.FM0        EQU       %00000000           Single density track 0
DNS.MFM0       EQU       %00000100           Double density track 0

* Floppy disk step rate definitions
STP.30ms       EQU       0                   30ms step rate
STP.20ms       EQU       1                   20ms step rate
STP.12ms       EQU       2                   12ms step rate
STP.6ms        EQU       3                   6ms step rate

               PAG       
*************************************
* Random Block Path Descriptor Format
*
* A path descriptor is created for every new path that is open
* via the I$Open system call (processed by IOMan).  Process
* descriptors track state information of a path.
*
               ORG       PD.FST
PD.SMF         RMB       1                   State flags
PD.CP          RMB       4                   Current logical byte position
PD.SIZ         RMB       4                   File size
PD.SBL         RMB       3                   Segment beginning lsn
PD.SBP         RMB       3                   Segment beginning psn
PD.SSZ         RMB       3                   Segment size
PD.DSK         RMB       2                   Disk id
PD.DTB         RMB       2                   Drive table ptr
               ORG       PD.OPT
               RMB       1                   Device type
PD.DRV         RMB       1                   Drive number
PD.STP         RMB       1                   Step rate
PD.TYP         RMB       1                   Disk device type (5" 8" other)
PD.DNS         RMB       1                   Density capability
PD.CYL         RMB       2                   Number of cylinders
PD.SID         RMB       1                   Number of surfaces
PD.VFY         RMB       1                   0=verify disk writes
PD.SCT         RMB       2                   Default sectors/track
PD.T0S         RMB       2                   Default sectors/track tr00,s0
PD.ILV         RMB       1                   Sector interleave offset
PD.SAS         RMB       1                   Segment allocation size
PD.TFM         RMB       1                   DMA Transfer Mode
PD.Exten       RMB       2                   Path Extension (PE) for record locking
PD.SToff       RMB       1                   Sector/Track offsets (for "foreign" disk formats)
PD.ATT         RMB       1                   File attributes
PD.FD          RMB       3                   File descriptor psn
PD.DFD         RMB       3                   Directory file descriptor psn
PD.DCP         RMB       4                   File directory entry ptr
PD.DVT         RMB       2                   User readable dev tbl ptr

* State Flags
BUFMOD         EQU       $01                 Buffer modified
SINBUF         EQU       $02                 Sector in buffer
FDBUF          EQU       $04                 File descriptor in buffer
*EOFSEC equ $08 End of file sector
*EOF equ $10 End of file
InDriver       EQU       $20                 Currently in Disk Driver, or queued
BufBusy        EQU       $40                 Buffer is currently busy

               IFNE      Level-1
************************************
* Random Block Path Extension Format
*
* RBF paths under Level Two have additional information that
* is referenced by the path extension area.
*
               ORG       0
PE.PE          RMB       1                   PE path number
PE.PDptr       RMB       2                   Back ptr to this PE's Path Descriptor
PE.NxFil       RMB       2                   Drive Open-File list ptr
PE.Confl       RMB       2                   Circular File Conflict list
PE.Lock        RMB       1                   Path lockout status
PE.LoLck       RMB       4                   Low Locked Logical addr
PE.HiLck       RMB       4                   High Locked Logical addr
PE.Wait        RMB       2                   PE ptr to (next) locked-out PE
PE.TmOut       RMB       2                   Max ticks to wait for locked segment
PE.Owner       RMB       1                   Process ID of owner of locked segment
PE.Req         RMB       1                   Temp for PE.Lock in GAIN when LockSeg fails
PE.Prior       RMB       1                   Temp for process priority while in driver
PE.SigSg       RMB       1                   Signal code to send
PE.SigID       RMB       1                   Process ID to send the signal to
               RMB       32-.                Reserved
PE.FilNm       RMB       32                  Temp for filename during directory search

* PE.Lock status codes
Unlocked       EQU       0                   No portion of file is locked
RcdLock        EQU       1                   Record from LoLck to HiLck locked
FileLock       EQU       2                   Entire file locked
EofLock        EQU       4                   End of file is locked
               ENDC      


               PAG       
***********************
* LSN0 Disk Data Format
*
* Logical Sector Number 0 is the first sector on an RBF formatted device
* and contains information about the device's size and format.
*
               ORG       0
DD.TOT         RMB       3                   Total number of sectors
DD.TKS         RMB       1                   Track size in sectors
DD.MAP         RMB       2                   Number of bytes in allocation bit map
DD.BIT         RMB       2                   Number of sectors/bit
DD.DIR         RMB       3                   Address of root directory fd
DD.OWN         RMB       2                   Owner
DD.ATT         RMB       1                   Attributes
DD.DSK         RMB       2                   Disk ID
DD.FMT         RMB       1                   Disk format; density/sides
DD.SPT         RMB       2                   Sectors/track
DD.RES         RMB       2                   Reserved for future use
DD.SIZ         EQU       .                   Device descriptor minimum size
DD.BT          RMB       3                   System bootstrap sector
DD.BSZ         RMB       2                   Size of system bootstrap
DD.DAT         RMB       5                   Creation date
DD.NAM         RMB       32                  Volume name
DD.OPT         RMB       32                  Option area

*************************
* 2012/11/09 Gene Heskett - Some additions for bootlink etc use.
		ORG	$60			More options from device descriptor
VD.STP		RMB	1	This is for the /sh and /ih virtual disks in HDBDOS
VD.OFS		RMB	3	This is DT.WPC and DT.OSF combined, might be more 

* DD.FMT Bit Definitions - valid only if device is a floppy disk
FMT.SIDE       EQU       %00000001           Single Sided=0, Double Sided=1
FMT.DNS        EQU       %00000010           Single Density=0, Double Density=1
FMT.TDNS       EQU       %00000100           Track Density: 48/135 TPI=0, 96 TPI=1
FMT.T0DN       EQU       %00100000           Track 0 Density, see FMT.DNS


               PAG       
************************
* File Descriptor Format
*
* The file descriptor is a sector that is present for every file
* on an RBF device.  It contains attributes, modification dates,
* and segment information on a file.
*
               ORG       0
FD.ATT         RMB       1                   Attributes
FD.OWN         RMB       2                   Owner
FD.DAT         RMB       5                   Date last modified
FD.LNK         RMB       1                   Link count
FD.SIZ         RMB       4                   File size
FD.Creat       RMB       3                   File creation date (YY/MM/DD)
FD.SEG         EQU       .                   Beginning of segment list
* Segment List Entry Format
               ORG       0
FDSL.A         RMB       3                   Segment beginning physical sector number
FDSL.B         RMB       2                   Segment size
FDSL.S         EQU       .                   Segment list entry size
FD.LS1         EQU       FD.SEG+((256-FD.SEG)/FDSL.S-1)*FDSL.S
FD.LS2         EQU       (256/FDSL.S-1)*FDSL.S
MINSEC         SET       16


               PAG       
************************
* Directory Entry Format
*
* Directory entries are part of a directory and define the name
* of the file, as well as a pointer to its file descriptor.
*
               ORG       0
DIR.NM         RMB       29                  File name
DIR.FD         RMB       3                   File descriptor physical sector number
DIR.SZ         EQU       .                   Directory record size


               PAG       
********************
* RBF Static Storage
*
* Overall Disk Static Storage
*
* Note:  This does not reserve any memory for drive tables.  Each
*        driver is responsible for reserving sufficient memory for
*        the appropriate number of tables.
*
               ORG       V.USER              Reserve required           ($06)
V.NDRV         RMB       1                   Number of drives           ($07)
               RMB       8                   reserved                   ($08)
DRVBEG         EQU       .                   Beginning of drive tables  ($10)


               PAG       
****************
* Global Storage For Disk Drive Tables
*
* Each table contains the first 'DD.SIZ' bytes from
* LSN 0, and the current track, stepping rate,
* bitmap use flag, and disk type.
*
               ORG       0
               RMB       DD.SIZ              Device descriptor, LSN 0
V.TRAK         RMB       2                   Current track
V.BMB          RMB       1                   Bit-map use flag
V.FileHd       RMB       2                   Open file list for this drive
V.DiskID       RMB       2                   Disk ID
V.BMapSz       RMB       1                   Bitmap Size
V.MapSct       RMB       1                   Lowest reasonable bitmap sector
V.ResBit       RMB       1                   Reserved bitmap sector (for compaction)
V.ScTkOf       RMB       1                   Sector/Track byte (Combined from descriptor)
V.ScOfst       RMB       1                   Sector offset split from byte above
V.TkOfst       RMB       1                   Track offset split from byte above
               RMB       4                   Reserved
DRVMEM         EQU       .

               ENDC      

